storage/innobase/innodb.cmake

```diff
@@ -116,11 +116,18 @@ OPTION(WITH_INNODB_AHI "Include innodb_adaptive_hash_index" ON)
 OPTION(WITH_INNODB_ROOT_GUESS "Cache index root block descriptors" ON)
 IF(WITH_INNODB_AHI)
   ADD_DEFINITIONS(-DBTR_CUR_HASH_ADAPT -DBTR_CUR_ADAPT)
+  add_feature_info(INNODB_AHI "ON" "InnoDB Adaptive Hash Index")
+  add_feature_info(INNODB_ROOT_GUESS "ON" "Cache index root block descriptors in InnoDB")
   IF(NOT WITH_INNODB_ROOT_GUESS)
     MESSAGE(WARNING "WITH_INNODB_AHI implies WITH_INNODB_ROOT_GUESS")
   ENDIF()
 ELSEIF(WITH_INNODB_ROOT_GUESS)
+  add_feature_info(INNODB_AHI "OFF" "InnoDB Adaptive Hash Index")
+  add_feature_info(INNODB_ROOT_GUESS "ON" "Cache index root block descriptors in InnoDB")
   ADD_DEFINITIONS(-DBTR_CUR_ADAPT)
+ELSE()
+  add_feature_info(INNODB_AHI "OFF" "InnoDB Adaptive Hash Index")
+  add_feature_info(INNODB_ROOT_GUESS "OFF" "Cache index root block descriptors in InnoDB")
 ENDIF()
 
 OPTION(WITH_INNODB_EXTRA_DEBUG "Enable extra InnoDB debug checks" OFF)
```

Comment:
Only one add_feature_info(INNODB_AHI...) and one add_feature_info(INNODB_ROOT_GUESS) would suffice